indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -4381,6 +4374,39 @@ public ConcurrentHashMap<PartitionIdType, SequenceOffsetType> getPartitionOffset
     return partitionOffsets;
   }
 
+  /**
+   * Calculates the average poll-idle-ratio metric across all active tasks.
+   * This metric indicates how much time the consumer spends idle waiting for data.
+   *
+   * @return the average poll-idle-ratio across all tasks, or 1 (full idle) if no tasks or metrics are available
```

Comment:
```suggestion
   * @return the average poll-idle-ratio across all tasks, or 0 (fully busy) if no tasks or metrics are available
```

Reply:
On second though, we might want to return -1 from here if we don't have any metrics available.
This would cause the auto-scaler to skip scaling rather than make a bad decision.